soroban-env-host/src/host.rs

```diff
@@ -819,7 +818,35 @@ impl Host {
                 } else {
                     Err(self.err_status_msg(
                         ScHostFnErrorCode::InputArgsWrongLength,
-                        "unexpected arguments to 'CreateContractWithSource' host function",
+                        "unexpected arguments to 'CreateContractWithSourceAccount' host function",
+                    ))
+                }
+            }
+            HostFunction::CreateTokenContractWithAsset => {
+                if let [ScVal::Object(Some(a_obj))] = args.as_slice() {
```

Comment:
There's no validation here of the type of a_obj, does that matter? I'm guessing we validate the type within create_token_wrapper.

Reply:
create_token_wrapper will try to deserialize the byte vector into an xdr::Asset, and error if it can't.